server/auditlog_util/auditlog_util.go

```diff
@@ -0,0 +1,26 @@
+package auditlog_util
```

Comment:
naming something `util` is always tricky, because it's hard to tell what belongs in util and what belongs in auditlog.

what if we just added 3 methods to auditlog

`al.LogWithResource`
`al.LogWithSecret`
`al.LogWithInvocation`

alternatively, we could just remove these methods altogether and either have `al.Log` take a type and a proto i.e.
```
al.Log(ctx, auditlog_util.InvocationResourceID(req.GetInvocationId()), alpb.Action_UPDATE, req)
```
becomes (which is actually less characters)
```
al.Log(ctx, alpb.ResourceType_INVOCATION, req.GetInvocationId(), alpb.Action_UPDATE, req)
```

Reply:
Taking a type and a string ID won't work for the more elaborate resource types such as Group API Key that requires more than ID to initialize resource. For example, this is currently being used

```go
		rid := &alpb.ResourceID{
			Type: alpb.ResourceType_GROUP_API_KEY,
			Id:   req.GetApiKeyId(),
			Name: key.Label,
		}
		al.Log(ctx, rid, alpb.Action_ACCESS, req)
```

I added 1 new method `al.LogWithResourceType` instead.